### PR TITLE
Don't pad 12h timestamps

### PR DESCRIPTION
**Pull request by turt2live**
As per @lampholder's comment [here](https://github.com/vector-im/riot-web/issues/4393#issuecomment-312266849), this is the desirable behaviour.

Changes `03:00 PM` to be `3:00 PM`.

### CODE REVIEW

File: src/DateUtils.js [hunk 1]
```diff
@@ -55,7 +55,6 @@ function twelveHourTime(date) {
     let hours = date.getHours() % 12;
     const minutes = pad(date.getMinutes());
     const ampm = date.getHours() >= 12 ? 'PM' : 'AM';
-    hours = pad(hours ? hours : 12);
```
Comment (t3chguy) [inline src/DateUtils.js]: surely the `hours ? hours : 12` must now go into the template literal below? Or am I too tired

Comment (turt2live) [inline src/DateUtils.js]: @t3chguy good point, fixed.

File: src/DateUtils.js [hunk 2]
```diff
@@ -55,7 +55,7 @@ function twelveHourTime(date) {
     let hours = date.getHours() % 12;
     const minutes = pad(date.getMinutes());
     const ampm = date.getHours() >= 12 ? 'PM' : 'AM';
```
Comment (lukebarnard1) [inline src/DateUtils.js]: whilst you're here, we should do i18n for 'PM'/'AM'. Apparently Korean is an example of a language that uses something else.

Comment (turt2live) [inline src/DateUtils.js]: @lukebarnard1 added

**Pull request merged**